### ISSUE TITLE
smithy-client should use ^2.2.1 starting from release 3.484.0 because classBuilder commit not included in 2.2.0

### ISSUE THREAD

**Issue by peng-shi-mck** [aws/aws-sdk-js-v3#5638]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

![image](https://github.com/aws/aws-sdk-js-v3/assets/116387693/2531c7d5-b677-4a4d-99e4-4141639cc23f)

https://github.com/aws/aws-sdk-js-v3/blob/v3.484.0/clients/client-secrets-manager/package.json

### SDK version number

@aws-sdk/client-secrets-manager:3.484.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

node 16

### Reproduction Steps

compile one application and deploy to AWS lambda

### Observed Behavior

2024-01-03T20:13:30.951Z	undefined	ERROR	Uncaught Exception 	
{
    "errorType": "TypeError",
    "errorMessage": "smithy_client_1.Command.classBuilder is not a function",
    "stack": [
        "TypeError: smithy_client_1.Command.classBuilder is not a function",
        "    at Object.<anonymous> (/opt/node_modules/@aws-sdk/client-secrets-manager/dist-cjs/commands/BatchGetSecretValueCommand.js:12:6)",
        "    at Module._compile (node:internal/modules/cjs/loader:1198:14)",
        "    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)",
        "    at Module.load (node:internal/modules/cjs/loader:1076:32)",
        "    at Function.Module._load (node:internal/modules/cjs/loader:911:12)",
        "    at Module.require (node:internal/modules/cjs/loader:1100:19)",
        "    at require (node:internal/modules/cjs/helpers:119:18)",
        "    at Object.<anonymous> (/opt/node_modules/@aws-sdk/client-secrets-manager/dist-cjs/SecretsManager.js:5:38)",
        "    at Module._compile (node:internal/modules/cjs/loader:1198:14)",
        "    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)"
    ]

### Expected Behavior

no error

### Possible Solution

change dep from ^2.2.0 to ^2.2.1 for "@smithy/smithy-client": "^2.2.0" in https://github.com/aws/aws-sdk-js-v3/blob/v3.484.0/clients/client-secrets-manager/package.json

**Comment by RanVaknin** [comment]
Hi @peng-shi-mck,

Unfortunately I'm not able to reproduce the error.

Example:
```js
const { SecretsManagerClient, GetSecretValueCommand } = require('@aws-sdk/client-secrets-manager');

const client = new SecretsManagerClient({region: "us-east-1"})

async function foo() {
    try {
        const res = await client.send(new GetSecretValueCommand({ 
            SecretId: "my-secret"
        }));
        console.log(res)
    } catch (error) {
        console.log(error)
    }
}
foo();
```

```console
$ node -v
v16.13.2
$ npm ls
test_folder/4494-secrets-manager
└── @aws-sdk/client-secrets-manager@3.484.0

$ node index.js 
{
  '$metadata': {
    httpStatusCode: 200,
    requestId: 'REDACTED',
    extendedRequestId: undefined,
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  ARN: 'arn:aws:secretsmanager:us-east-1:REDACTED:secret:my-secret-uoysFh',
  CreatedDate: 2022-06-02T22:14:54.561Z,
  Name: 'my-secret-3425',
  SecretString: '{"username":"REDACTED","password":"REDACTED","engine":"REDACTED","host":"test-REDACTED.cluster-REDACTED.us-east-1.rds.amazonaws.com","port":REDACTED,"dbClusterIdentifier":"REDACTED"}',
  VersionId: 'REDACTED',
  VersionStages: [ 'AWSCURRENT' ]
}
```

If I had to guess, you probably have another dependency in your tree relying on an older version of `@aws-sdk/client-secrets-manager`. You can check that by running `npm ls @aws-sdk/client-secrets-manager`. That will generate a tree of all your dependencies and will pinpoint the ones that rely on the secrets manager client. If there's a mismatch, my suggestion is to let npm resolve the dependency tree on its own by doing:

```bash
cd your/project/root
rm package-lock.json
rm -rf node_modules
npm install
```

Let me know how it goes.
Ran~

**Comment by peng-shi-mck** [comment]
Thank you for looking into this, @RanVaknin 

Before we had 
```
"@aws-sdk/client-secrets-manager": "^3.418.0
```
and we don't have package-lock.
Build applied 3.484.0 and we hit the runtime error above in the description.

client-secret-mananger@3.484.0 requires classBuilder which was submit via commit 2df620b(tagged with ^2.2.1 in smithy-client).
meanwhile the dep version for smithy-client is set to ^2.2.0 for client-secret-manager@3.484.0, and I think this is the reason we run into the error.

I fixed error in our env by set 
```
"@aws-sdk/client-secrets-manager": "^3.418.0 <3.478.0"
```

BRs,
Percy

**Comment by kuhe** [comment]
https://github.com/smithy-lang/smithy-typescript/commit/2df620b is the Java code generation commit of classBuilder and has no effect on the JavaScript runtime packages.

classBuilder was released in `@smithy/smithy-client`@2.2.0 as can be seen here: https://github.com/smithy-lang/smithy-typescript/releases/tag/%40smithy%2Fsmithy-client%402.2.0
